lib/rules/no-test-support-import.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @fileOverview no importing of test files and no exporting in test files
+ */
+
+'use strict';
+
+const NO_IMPORT_MESSAGE =
+  'Do not import a file from test-support into production code, only into files located in test.';
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow importing of "test-support" files in production code.',
+      category: 'Testing',
+      recommended: true,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-test-support-import.md',
+    },
+    schema: [],
+    importMessage: NO_IMPORT_MESSAGE,
+  },
+
+  create: function create(context) {
+    const fileName = context.getFilename();
+
+    return {
+      ImportDeclaration(node) {
+        const importSource = node.source.value;
+
+        if (
+          importSource.includes('test-support') &&
+          !(fileName.includes('/tests/') || fileName.includes('/addon-test-support/'))
```

Comment:
I think you also need to add a check for `fileName.includes('/test-support/')` (that is another valid location for addons to put test specific content).

Reply:
added this check as well as another test.